parsers/callArgsParser.go

```diff
@@ -32,6 +34,24 @@ func (parser *callArgsParser) ParseData(data string) (string, [][]byte, error) {
 	return function, arguments, nil
 }
 
+// ParseArguments parses strings of the following format:
+// argFoo@hex(argBarHex)...
+func (parser *callArgsParser) ParseArguments(data string) ([][]byte, error) {
+	tokens := strings.Split(data, atSeparator)
+	arguments := make([][]byte, 0)
+	arguments = append(arguments, []byte(tokens[0]))
+	for i := minNumCallArguments; i < len(tokens); i++ {
```

Comment:
instead of this for loop we can call:
```GO
parsedArgs, err := parser.parseAguments(tokens)
if err != nil{
    return nil, err
}
arguments = append(arguments, parsedArgs...)
```

in this way we avoid the duplicated code

Reply:
done